core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -17,247 +17,342 @@
+        param_specs: returnDict.exploration.param_specs,
+        states: returnDict.exploration.states,
+        title: returnDict.exploration.title,
+        language_code: returnDict.exploration.language_code,
+        version: returnDict.version
       },
-      get onOppiaFeedbackAvailable() {
-        return _oppiaFeedbackAvailableEventEmitter;
+      returnDict.version,
+      returnDict.preferred_audio_language_code,
+      returnDict.auto_tts_enabled,
+      returnDict.preferred_language_codes,
+      arePretestsAvailable ? () => {} : callback);
+  }
+
+  initializePretestServices(
+      pretestQuestionDicts: QuestionBackendDict[],
+      callback: (
+        initialCard: StateCard, nextFocusLabel: string) => void): void {
+    this.playerCorrectnessFeedbackEnabledService.init(true);
+    this.questionPlayerEngineService.init(
+      pretestQuestionDicts, callback, () => {});
+  }
+
+  initializeQuestionPlayerServices(
+      questionDicts: QuestionBackendDict[],
+      successCallback: (initialCard: StateCard, nextFocusLabel: string) => void,
+      errorCallback: () => void): void {
+    this.playerCorrectnessFeedbackEnabledService.init(true);
+    this.questionPlayerEngineService.init(
+      questionDicts, successCallback, errorCallback);
+  }
+
+  setExplorationMode(): void {
+    this.explorationMode = ExplorationPlayerConstants
+      .EXPLORATION_MODE.EXPLORATION;
+    this.currentEngineService = this.explorationEngineService;
+  }
+
+  setPretestMode(): void {
+    this.explorationMode = ExplorationPlayerConstants.EXPLORATION_MODE.PRETEST;
+    this.currentEngineService = this.questionPlayerEngineService;
+  }
+
+  setQuestionPlayerMode(): void {
+    this.explorationMode = ExplorationPlayerConstants
+      .EXPLORATION_MODE.QUESTION_PLAYER;
+    this.currentEngineService = this.questionPlayerEngineService;
+  }
+
+  setStoryChapterMode(): void {
+    this.explorationMode = ExplorationPlayerConstants
+      .EXPLORATION_MODE.STORY_CHAPTER;
+    this.currentEngineService = this.explorationEngineService;
+  }
+
+  initExplorationPreviewPlayer(
+      callback: (sateCard: StateCard, str: string) => void): void {
+    this.setExplorationMode();
+    Promise.all([
+      this.editableExplorationBackendApiService.fetchApplyDraftExplorationAsync(
+        this.explorationId),
+      this.explorationFeaturesBackendApiService.fetchExplorationFeaturesAsync(
+        this.explorationId),
+    ]).then((combinedData) => {
+      let explorationData = combinedData[0];
+      let featuresData: ExplorationFeatures = combinedData[1];
+
+      this.explorationFeaturesService.init({
+        param_changes: explorationData.param_changes,
+        states: explorationData.states
+      }, featuresData);
+      this.explorationEngineService.init(
+        explorationData, null, null, null, null, callback);
+      this.playerCorrectnessFeedbackEnabledService.init(
+        explorationData.correctness_feedback_enabled);
```

Comment:
`correctness_feedback_enabled` property did not exist on ExplorationBackendDict interface earlier. I have added it now.